main.hpp

```diff
@@ -51,162 +56,190 @@
+	    {
+	    	data_names.push_back(it->first);
+	    }
+
+	    debugfile << "i,\tj,\tk,\t";
+	    for (string item : data_names)
+	    {
+	    	debugfile << item << "\t";
+	    }
+
+	    debugfile << "-- repeated for all segments" << endl;
+
+		for (auto block : well_indices[well_name])
+		{
+		    debugfile << block.ijk_index().i() + 1 << "\t" << block.ijk_index().j() + 1 << "\t" << block.ijk_index().k() + 1 << "\t";
+
+		    calc_data = block.get_calculation_data();
+
+	    	for (int iSegment = 0; iSegment < calc_data[data_names[0]].size(); ++iSegment)
+			{
+			    for (string item : data_names)
+			    {
+			    	debugfile << calc_data[item].at(iSegment) << "\t";
+			    }
+			}
+
+	    	debugfile << endl;
+		}
+    }
+
+    debugfile.close();
+}
+
+po::variables_map createVariablesMap(int argc, const char **argv) 
+{
+    //
+    // This function parses the runtime arguments and creates a boost::program_options::variable_map from them.
+    // It also displays help if the --help flag is passed.
+    //
+	
+    po::options_description desc("WellIndexCalc options");
+    desc.add_options()
+            ("help", "print help message")
+
+            ("debug", po::value<int>()->implicit_value(0), "write debug information")
+
+            ("grid,g", po::value<string>()->required(), "path to model grid file (e.g. *.GRID)")
+
+            ("well-filedef,f", po::value<string>(), "path to the well(s) definition file name")
+
+            ("heel,h", po::value<vector<double>>()->multitoken(), "Heel coordinates (x y z)")
+
+            ("toe,t", po::value<vector<double>>()->multitoken(), "Toe coordinates (x y z)")
+
+            ("radius,r", po::value<double>(), "wellbore radius")
+
+            ("compdat,c", po::value<int>()->implicit_value(0), "print in compdat format instead of CSV")
+
+            ("well-name,w", po::value<string>(), "well name to be used when writing compdat")
+            ;
+	
+    // Process arguments to variable map
+    po::variables_map vm;
+    
+    bool success_arg = true;
+    try
+    {
+		// Parse the input arguments and store the values 
+		po::store(po::parse_command_line(argc, argv, desc, 
+				po::command_line_style::unix_style ^ po::command_line_style::allow_short), vm);
+		
+		// ??
+		po::notify(vm);	
+    } 
+    catch (std::exception& e) 
+    {
+    	success_arg = false;
+    	std::cout << e.what() << std::endl;
+    }
+    
+    // If called with --help or -h flag:
+    if (vm.count("help") || !success_arg) 
+    { // Print help if --help present or input file/output dir not present
+        cout << "Usage: ./WellIndexCalc --grid gridpath --heel x1 y1 z1 --toe x2 y2 z2 --radius r [options]" << endl;
+        cout << "options can be --compdat --well-name Name" << endl;
+        cout << "Or" << endl;
+        cout << "Usage: ./WellIndexCalc --grid gridpath --well-filedef filepath" << endl;
+        cout << desc << endl;
+        
```

Comment:
I think in this loop we should add the debug code above that tells us specifically which parameter that are missing/wrong